### PR TITLE
Implements addmod, addexp, deletemod, deleteexp functions and fixes bug to update actual workload

### LINKED ISSUE #38: Implement function to add a module
This function will check if the module input by the user exists. If the module does not exist, this function will create a module and add the module to the list of modules.

### LINKED ISSUE #39: Implement function to add module with expected time
This function will check if the module exists. If the module does not exist, this function will creates and adds a modue with expected time to the list of modules. If module already exist, this function will update expected time based on user input.

### LINKED ISSUE #40: Implement function to delete expected time of module
This function will check if the module exists. If module exists, this function will delete the expected time of the module.

### LINKED ISSUE #41: Implement function to delete module
This function will check if the module exists. If module exists, this function will delete the module from the list of modules.

### LINKED ISSUE #42: Actual workload of module is not updated
The index of the module in the modList is not returned correctly. The newly created module and the module that was in the modList are different objects.

### PR DESCRIPTION
**Pull request by KennethEer**
Implements a function to add module.  
Implements a function to add module with expected workload. 
Implements a function to delete module. 
Implements a function to delete expected workload of module.
Fixes the bug to update the actual time spent by the user. 

This closes #38, closes #39, closes #40, closes #41, and closes #42

### CODE REVIEW

File: src/main/java/seedu/modtracker/Module.java [hunk 1]
```diff
@@ -51,4 +55,22 @@ public void minusActualTime(String time, String week) {
...
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof Module)) {
+            return false;
+        }
+        Module m = (Module) obj;
+
+        return moduleCode.equals(m.moduleCode);
+    }
```
Comment (kerct) [inline src/main/java/seedu/modtracker/Module.java:74]: Why not just take in Module as an argument?

File: src/main/java/seedu/modtracker/ModuleList.java [hunk 2]
```diff
@@ -2,37 +2,121 @@
...
+        }
+    }
+
+    /**
+     * Checks if the module exists in the list of modules.
+     *
+     * @param input module code typed in by user
+     * @return the existence of module in the list of modules.
+     */
+    public boolean checkModuleIfExist(String input) {
```
Comment (kerct) [inline src/main/java/seedu/modtracker/ModuleList.java:94]: Is "checkIfModuleExist" a better name?

**Review by kerct — COMMENTED** [review]
Some minor comments but looks good

File: src/main/java/seedu/modtracker/Module.java (hunk 1 above)
Comment (KennethEer) [reply src/main/java/seedu/modtracker/Module.java:74]: The default `Object.equals()` method takes in the object of Object type as the parameter and I keep this method signature to overwrite the default method.

File: src/main/java/seedu/modtracker/ModuleList.java (hunk 2 above)
Comment (KennethEer) [reply src/main/java/seedu/modtracker/ModuleList.java:94]: Yes, the function name is renamed to checkIfModuleExist.

**Pull request merged**